src/main/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/CreateCollectionCommandResolver.java

```diff
@@ -319,22 +343,60 @@ private void validateAuthentication(
     }
   }
 
+  /**
+   * Validates the parameters provided by the user against the expected parameters from both the
+   * provider and the model configurations. This method ensures that only configured parameters are
+   * provided, all required parameters are included, and no unexpected parameters are passed.
+   *
+   * @param userConfig The vector search configuration provided by the user.
+   * @param providerConfig The configuration of the embedding provider which includes model and
+   *     provider-level parameters.
+   * @throws JsonApiException if any unconfigured parameters are provided, required parameters are
+   *     missing, or if an error occurs due to no parameters being configured but some are provided
+   *     by the user.
+   */
   private void validateUserParameters(
       CreateCollectionCommand.Options.VectorSearchConfig.VectorizeConfig userConfig,
       EmbeddingProvidersConfig.EmbeddingProviderConfig providerConfig) {
+    // 0. Combine provider level and model level parameters
+    List<EmbeddingProvidersConfig.EmbeddingProviderConfig.ParameterConfig> allParameters =
+        new ArrayList<>();
+    // Add all provider level parameters
+    allParameters.addAll(providerConfig.parameters());
+    // Get all the parameters except "vectorDimension" for the model -- model has been validated in
+    // the previous step
+    List<EmbeddingProvidersConfig.EmbeddingProviderConfig.ParameterConfig> modelParameters =
+        providerConfig.models().stream()
+            .filter(m -> m.name().equals(userConfig.modelName()))
+            .findFirst()
+            .map(EmbeddingProvidersConfig.EmbeddingProviderConfig.ModelConfig::parameters)
+            .map(
+                params ->
+                    params.stream()
+                        .filter(
+                            param ->
+                                !param
+                                    .name()
+                                    .equals(
+                                        "vectorDimension")) // Exclude 'vectorDimension' parameter
+                        .collect(Collectors.toList()))
+            .get();
+    // Add all model level parameters
+    allParameters.addAll(modelParameters);
+
     // 1. Error if the user provided unconfigured parameters
-    if (providerConfig.parameters() == null || providerConfig.parameters().isEmpty()) {
-      // If providerConfig.parameters() is null or empty but the user still provides parameters,
-      // it's an error
+    if (allParameters.isEmpty()) {
+      // If allParameters is empty but the user still provides parameters, error out
       if (userConfig.parameters() != null && !userConfig.parameters().isEmpty()) {
         throw ErrorCode.INVALID_CREATE_COLLECTION_OPTIONS.toApiException(
```

Comment:
Is this good message? It seems different from intent here -- should probably say user is providing extra parameters, and also list names of "extra" ones to help troubleshoot?

Reply:
Sure. After thinking about it, I decided to remove this if statement. If user provides extra parameters, the error message will be `"Unexpected parameter '%s' for the provider '%s' provided",
                    userParamName, userConfig.provider()`

Reply:
Makes sense.